src/Form/OembedDialog.php

```diff
@@ -0,0 +1,807 @@
+    $entity_type = $entity_element['data-entity-type'];
+    $entity = $uuid && $entity_type ? [$this->entityRepository->loadEntityByUuid($entity_type, $uuid)] : [];
+    $form_state->set('entity', current($entity) ?: NULL);
+
+    if (!$form_state->get('step')) {
+      // If an entity has been selected, then always skip to the embed options.
+      if ($form_state->get('entity')) {
+        $form_state->set('step', 'embed');
+      }
+      else {
+        $form_state->set('step', 'select');
+      }
+    }
+
+    $form['#tree'] = TRUE;
+    $form['#attached']['library'][] = 'editor/drupal.editor.dialog';
+    $form['#attached']['library'][] = 'oe_oembed/oe_oembed.dialog';
+    $form['#prefix'] = '<div id="oe-oembed-entity-dialog-form">';
+    $form['#suffix'] = '</div>';
+    $form['#attributes']['class'][] = 'oe-oembed-entity-dialog-step--' . $form_state->get('step');
+
+    $this->loadEntityBrowser($form_state);
+
+    if ($form_state->get('step') == 'select') {
+      $form = $this->buildSelectStep($form, $form_state);
+    }
+    elseif ($form_state->get('step') == 'review') {
+      $form = $this->buildReviewStep($form, $form_state);
+    }
+    elseif ($form_state->get('step') == 'embed') {
+      $form = $this->buildEmbedStep($form, $form_state);
+    }
+
+    return $form;
+  }
+
+  /**
+   * Builds the form of the selection step.
+   *
+   * @param array $form
+   *   An associative array containing the structure of the form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
+   *
+   * @return array
+   *   The form structure.
+   */
+  public function buildSelectStep(array &$form, FormStateInterface $form_state): array {
+    // Entity element is calculated on every AJAX request/submit.
+    // See self::buildForm().
+    $entity_element = $form_state->get('entity_element');
+    /** @var \Drupal\Core\Entity\ContentEntityInterface $entity */
+    $entity = $form_state->get('entity');
+    /** @var \Drupal\embed\EmbedButtonInterface $embed_button */
+    $embed_button = $form_state->get('embed_button');
+    $entity_type = $embed_button->getTypeSetting('entity_type');
+
+    $form['#title'] = $this->t('Select an entity to embed');
+
+    if ($this->entityBrowser) {
+      $this->eventDispatcher->addListener(Events::REGISTER_JS_CALLBACKS, [
+        $this,
+        'registerJsCallback',
+      ]);
+      $form['entity_browser'] = [
+        '#type' => 'entity_browser',
+        '#entity_browser' => $this->entityBrowser->id(),
+        '#cardinality' => 1,
+        '#entity_browser_validators' => [
+          'entity_type' => ['type' => $entity_type],
+        ],
+      ];
+    }
+    else {
+      $bundles = $embed_button->getTypeSetting('bundles');
+      $form['entity_id'] = [
+        '#type' => 'entity_autocomplete',
+        '#target_type' => $entity_type,
+        '#selection_settings' => [
+          'target_bundles' => $bundles,
```

Comment:
Locally on unchecking all allowed bundles, I wasn't able to select created media with autocomplete.
I expected to have the possibility to search with autocomplete entities from all bundles.
Seems like at the end we should have NULL to be able to select entity from all bundles.
It can be fixed I believe in other places. Even better inside the plugin.
```suggestion
          'target_bundles' => $bundles ?? NULL,
```